rest-server/core/src/main/scala/com/actionml/core/spark/SparkMongoSupport.scala

```diff
@@ -17,18 +17,43 @@
 
 package com.actionml.core.spark
 
-import com.mongodb.spark.MongoSpark
+import com.actionml.core.store.backends.MongoStorage
+import com.mongodb.MongoClient
+import com.mongodb.client.MongoDatabase
+import com.mongodb.spark.{MongoClientFactory, MongoConnector, MongoSpark}
 import org.apache.spark.SparkContext
 import org.apache.spark.rdd.RDD
-import org.bson.Document
+import org.bson.codecs.configuration.CodecProvider
 
+import scala.reflect.ClassTag
 
-trait SparkStoreSupport[T] {
-  def readRdd(sc: SparkContext): RDD[T]
+
+trait SparkStoreSupport {
+  def readRdd[T](sc: SparkContext, codecs: List[CodecProvider])(implicit ct: ClassTag[T]): RDD[T]
+}
+
+trait SparkMongoSupport extends SparkStoreSupport {
+
+  override def readRdd[T](sc: SparkContext, codecs: List[CodecProvider] = List.empty)(implicit ct: ClassTag[T]): RDD[T] = {
```

Comment:
This should be in the DAO, shouldn't it? It should essentially be a `readMany` that returns an RDD, right? Allowing the consumer to pass query params like filters. This will be quite typical in real use.